pkg/provider/provider.go

```diff
@@ -571,49 +622,180 @@ func ConfigureProvider(ctx context.Context, s *schema.ResourceData) (any, diag.D
 		}
 	}
 
-	config := &gosnowflake.Config{
-		Application: "terraform-provider-snowflake",
+	config, err := getDriverConfigFromTerraform(s)
+	if err != nil {
+		return nil, diag.FromErr(err)
 	}
 
-	if v, ok := s.GetOk("account"); ok && v.(string) != "" {
-		config.Account = v.(string)
+	if v, ok := s.GetOk("profile"); ok && v.(string) != "" {
+		tomlConfig, err := getDriverConfigFromTOML(v.(string))
+		if err != nil {
+			return nil, diag.FromErr(err)
+		}
+		config = sdk.MergeConfig(config, tomlConfig)
 	}
 
-	// backwards compatibility until we can remove this
-	if v, ok := s.GetOk("username"); ok && v.(string) != "" {
-		config.User = v.(string)
-	}
+	client, clientErr := sdk.NewClient(config)
 
-	if v, ok := s.GetOk("user"); ok && v.(string) != "" {
-		config.User = v.(string)
+	// needed for tests verifying different provider setups
+	if os.Getenv(resource.EnvTfAcc) != "" && os.Getenv(string(testenvs.ConfigureClientOnce)) == "true" {
+		configuredClient = client
+		configureClientError = clientErr
+	} else {
+		configuredClient = nil
+		configureClientError = nil
 	}
 
-	if v, ok := s.GetOk("password"); ok && v.(string) != "" {
-		config.Password = v.(string)
+	if clientErr != nil {
+		return nil, diag.FromErr(clientErr)
 	}
 
-	if v, ok := s.GetOk("warehouse"); ok && v.(string) != "" {
-		config.Warehouse = v.(string)
+	return &provider.Context{Client: client}, nil
+}
+
+func getDriverConfigFromTOML(profile string) (*gosnowflake.Config, error) {
+	if profile == "default" {
```

Comment:
I've decided to remove this special handling. I don't think it brings any value.

Reply:
But if I see correctly, you have changed the behavior of the provider.

Previously, leaving the profile empty (setting it implicitly to `default`) would lead to initialization through `sdk.DefaultConfig` which does not fail if config does not exist - it returns an empty driver config instead. So, not having the toml at all would result in no error, and now it will result in an error for config not found.